armi/bookkeeping/db/database3.py

```diff
@@ -798,6 +796,42 @@ def load(
 
         return root
 
+    def loadReadOnly(
+        self,
+        cycle,
+        node,
+        cs=None,
+        bp=None,
+        statePointName=None,
+    ):
+        """Load a new reactor, in read-only mode from a DB at (cycle, node).
+
+        Case settings and blueprints can be provided, or read from the database. Providing these
+        can be useful for snapshot runs or when you want to change settings mid-simulation. Geometry
+        is read from the database.
+
+        Parameters
+        ----------
+        cycle : int
+            Cycle number
+        node : int
+            Time node. If value is negative, will be indexed from EOC backwards like a list.
+        cs : armi.settings.Settings, optional
+            If not provided one is read from the database
+        bp : armi.reactor.Blueprints, optional
+            If not provided one is read from the database
+        statePointName : str, optional
+            Statepoint name (e.g., "special" for "c00n00-special/")
+
+        Returns
+        -------
+        Reactor
+            The top-level object stored in the database; a Reactor.
+        """
+        r = self.load(cycle, node, cs, bp, statePointName, allowMissing=True)
+        makeParametersReadOnly(r)
```

Comment:
One last thing is that volume needs to either be set for each component (by calling `componet.getVolume()` or the `getVolume()` call needs to be modified to not modify the volume parameter. This needs to happen before the parameters are made read only.

Reply:
I think volume might be special, but I'm curious about linked components and areas as well. There might be several edge cases here that need to be checked explicitly on component modifications. The tests seem to demonstrate that setting a parameter directly that is in read only won't work, but we should check mesh related things as well.

Reply:
@john-science or @alexhjames did we test any edge cases?